db/user.go

```diff
@@ -221,3 +222,74 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to map a User.Role to all User.Email's for that course.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input.
+// The function returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// The function returns an error if a course cannot be opened and the input contains roles or a *.
+// The function will issue a warning if it's given a valid course and an invalid role.
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    emailSet := map[string]any{};
+    roleSet := map[string]any{};
+
+    // Iterate over every given string, checking for emails, roles, and *, which denotes all roles.
+    for _, email := range emails {
+        if (email == "") {
+            continue;
+        }
+
+        if (strings.Contains(email, "@")) {
+            emailSet[strings.ToLower(email)] = nil;
+        } else {
+            if (email == "*") {
+                allRoles := model.GetAllRoleStrings();
+                for role := range allRoles {
+                    roleSet[role] = nil;
+                }
+            } else {
+                roleSet[strings.ToLower(email)] = nil;
+            }
+        }
+    }
+
+    if (len(roleSet) > 0) {
+        if (backend == nil) {
```

Comment:
I think this should probably be moved up to the front like with most DB functions.

I can see an argument that this functionality doesn't necessarily require a DB (which you can try to make),
but even when sending emails we may log (which does want a database).